scripts/cfos/make_dataset_data_tables.py

```diff
@@ -31,47 +33,16 @@ def upload_dataset_table_to_workspace(tsv_filenames_list, workspace_name, worksp
     print("Success: All data tables have been loaded to workspace.")
 
 
-def modify_file_table_dataframe(file_table_df):
-    """File table needs additional modification."""
-
-    file_types = ["features", "matrix", "barcode", "summary"]  # four types of files
-    file_type_dataframes = []  # empty list to hold re-organized dataframe per file type
-
-    # for each file_path type, create a separate dataframe
-    for file_type in file_types:
-        # list of columns to subset + the additional file_type column
-        cols_to_subset = ["library_id", "donor_id", "DataModality", "biosample_id"]
-        cols_to_subset.append(f"{file_type}_file_path")
-
-        # make file type specific dataframe and rename column to generic name
-        file_type_df = file_table_df[cols_to_subset]
-        file_type_df = file_type_df.rename(columns={f"{file_type}_file_path": "file_path"})
-
-        # create the file_type column and the file_id column (as concatenation of other values)
-        # TODO: determine if concatenation of columns is unique or if other file identifier is required
-        # TODO: parsing file paths may not be a safe automation across datasets
-        file_type_df["file_type"] = file_type
-        file_type_df["file_id"] = file_type_df["file_type"] + "_" + file_type_df["donor_id"] + "_" + file_type_df["library_id"]
-
-        # add dataframe to list
-        file_type_dataframes.append(file_type_df)
-
-    # concatenate all the file_type dataframes into one and return
-    modified_file_table_df = pd.concat(file_type_dataframes)
-    return modified_file_table_df
-
-
 def generate_load_table_files(dataset_tables_dict):
     """Generate load table tsv files for each of the dataset tables."""
 
     # instantiate empty list to hold names of generated tsv load files
     tsv_filenames_list = []
 
     # for each table in dictionary
-    for table_name in dataset_tables_dict:
+    for table_name in list(dataset_tables_dict.keys()):
         table_df = dataset_tables_dict[table_name]
-        if table_name == "file":
-            table_df = modify_file_table_dataframe(table_df)
+            
         # change table name column to new name --> entity:table_name_id and set as index
         table_df = table_df.rename(columns={f"{table_name}_id": f"entity:{table_name}_id"})
```

Comment:
as discussed on Slack: rather than requiring the schema to have a `{table_name}_id` field, you could do what TDR does and have the schema specify which field should be used as the table's primary key, and then copy it to a new field called `{table_name}_id` (leaving the original field named as-is - yes this duplicates data but (a) consistent logic with TDR import to Terra and (b) avoids confusion if the end user is expecting that field named just so.

Reply:
Updated! Ran the code and it duplicated and designated the appropriate columns as index columns, renamed to entity:tablename_id